core/core-funcs.el

```diff
@@ -36,12 +36,18 @@
   (run-hooks 'text-mode-hook))
 
 (defun spacemacs//get-package-directory (pkg)
-  "Return the directory of PKG. Return nil if not found."
-  (let ((elpa-dir (file-name-as-directory package-user-dir)))
-    (when (file-exists-p elpa-dir)
-      (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
-             (dir (car (directory-files elpa-dir 'full pkg-match))))
-        (when dir (file-name-as-directory dir))))))
+  "Return the installed directory of PKG. Return `nil' if not found."
+  (let ((directories (cons package-user-dir
+                           (and dotspacemacs-use-shared-packages
+                                package-directory-list)))
+        elpa-dir dir)
+    (while (and (not dir) directories)
+      (setq elpa-dir (pop directories))
+      (when (file-exists-p elpa-dir)
+        (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
+               (dir (car (directory-files elpa-dir 'full pkg-match))))
+          (when dir (setq dir (file-name-as-directory dir))))))
+    dir))
```

Comment:
Following uses catch/throw.  If you like this one better, I can switch to using it.
It does not matter to me which one is used.

```
(defun spacemacs//get-package-directory (pkg)
  "Return the installed directory of PKG. Return `nil' if not found."
  (let ((directories (cons package-user-dir
                           (and dotspacemacs-use-shared-packages
                                package-directory-list)))
        (pkg-match (concat (symbol-name pkg) "-[0-9]+"))
        elpa-dir dir)
    (catch 'found
      (mapcar (lambda (elpa-dir)
                (when (file-exists-p elpa-dir)
                  (let ((dir (car (directory-files elpa-dir 'full pkg-match))))
                    (if dir
                        (throw 'found (file-name-as-directory dir))))))
              directories))))
```